### ISSUE TITLE
Unable to connect to local peer on localhost in example

### ISSUE THREAD

**Issue by pharazen** [maidsafe/crust#1142]
I'm attempting to use the example to connect to localhost, and every time I try I get ConnectFailure. I've updated url dependency from the example, and added some logging, but otherwise everything is from crust master. Specifically, it's this bit that's triggered in the terminate function:

```
 if !core.user_data().connections.contains_key(&self.their_id) {
  let _ = self.event_tx.send(Event::ConnectFailure(self.their_id, "User data connections doesn't contain their id key".to_string()));
}
```
I checked the key count for that connections object and it contains 0 keys
```
println!("\nKey count in user data connections: {}\n", core.user_data().connections.keys().count());
```
Returns 0.

Any ideas what could be the issue?

**Comment by pharazen** [comment]
Screenshot:
[https://i.imgur.com/LCMuUd0.png](url)

**Comment by douglascaetano** [comment]
Hi @pharazen, thanks for your report!

Have you seen this `"User data connections doesn't contain their id key"` error since the first try?

Also, @ustulation just merged a few hours ago the PR #1141 which should fix the `ConnectFailure` error on local networks. Could you please test again with the latest version?

**Comment by pharazen** [comment]
Awesome, it works now. Great job and thanks for the help!
